search/collector/topn.go

```diff
@@ -215,6 +222,15 @@ func (hc *TopNCollector) Collect(ctx context.Context, searcher search.Searcher,
 			}
 		}
 
+		skipped, err = hc.adjustDocumentMatch(searchContext, reader, next)
```

Comment:
let me see if i got this right, with regards to the pagination stuff, so essentially the score of the knn hits got from the presearch/runknnCollector are adjusted based on the tf-idf score in the second phase, and the topn collector which maintains a skip+size amount of doc matches finalizes the results at the end where the total score of the knn hits can spread across the range of 0-skip+size hits that are there. 

when the user tries to get the 2nd page we just slice out that portion of the 0-skip+size hits that are there and now the retrieved second page doesn't necessarily have k nearest neighbors but some number less than that (which i believe is the expected behaviour since the pagination etc. is based on the total final score of a doc match) 

is the above thinking correct?

Reply:
say the phase 1 gives 4 docs D1 D2 D3 D4 with scores 1 2 3 4

say tfidf alone gives 3 docs D1 D5 D6 with scores 4 2 3 

say it is page 1 
from = 0 
size = 2

youll increase the score of D1 to be 5 and put D1 D2 D3 D4 D5 D6 into a min heap of size 2 and youll get the top 2 = D1 with 5 and D4 with 4

in page 2 
from = 2 
size = 2

youll now keep a heap of size 4 and do the above same step for page 1 basically
youll get D1,5 D4,4 D3,3 D6,3 in it (docId,score style this is)

you dump the first 2 since from is 2
youll output D3 and D6 which is page 2
so ye pagination is total score thing not based on knn anol